orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/pipelines/UpsertProjectStage.groovy

```diff
@@ -45,13 +45,17 @@ class UpsertProjectStage implements StageDefinitionBuilder {
   @Component
   @VisibleForTesting
   public static class UpsertProjectTask implements Task {
-    @Autowired
+    @Autowired(required = false)
     Front50Service front50Service
 
     UpsertProjectTask() {}
 
     @Override
     TaskResult execute(Stage stage) {
+      if (!front50Service) {
+        throw new RuntimeException("Unable to modify projects, front50 has not been enabled. Fix this by setting front50.enabled: true")
```

Comment:
I'd argue state is internal to the system, and is someplace orca can arrive when something unexpected happens, while on the other hand orca with a specific config doesn't support certain operations.